src/main/java/com/pusher/client/crypto/nacl/SecretBoxOpener.java

```diff
@@ -33,19 +36,14 @@ a copy of this software and associated documentation files (the "Software"),
     private byte[] key;
 
     public SecretBoxOpener(byte[] key) {
-        // TODO: add a a little preconditions lib/class for that
-        if (key == null) throw new IllegalArgumentException("null key passed");
-        if (key.length != 32) {
-            throw new IllegalArgumentException("key should be 32B, is: " + key.length + "B");
-        }
+        checkNotNull(key, "null key passed");
+        checkArgument(key.length == 32, "key should be 32B, is: " + key.length + "B");
```

Comment:
```suggestion
        checkArgument(key.length == 32, "key length should be 32 bytes, but is: " + key.length + " bytes");
```

Reply:
not sure what was wrong with the concise form, but didn't mind much the longer one so done that: dbdea4e53d468051462e3e7480e7a0a5368d174d just couldn't apply from GH as the 100 char limit was broken